lib/ssz_ex.ex

```diff
@@ -72,6 +85,103 @@ defmodule LambdaEthereumConsensus.SszEx do
     |> flatten_results_by(&Enum.join/1)
   end
 
+  defp decode_bitlist(bit_list, max_size) do
+    num_bytes = byte_size(bit_list)
+    num_bits = num_bytes * 8
+    padding = num_bits - bit_size(bit_list)
+    formatted_bit_list = <<0::size(padding), bit_list::bitstring>>
```

Comment:
This padding is not correctly calculated. bit_size(bit_list) will be exactly equal to num_bits, as we always decode full lists.

The first thing the decode function should do is get the trailing bit and use that for the bit length.